### ISSUE TITLE
[14.0] fastapi: json post requests support

### ISSUE THREAD

**Issue by destroT** [OCA/rest-framework#351]
**Is your feature request related to a problem?**
Fastapi does not accept json requests like in the 16.0 version. 

> werkzeug.exceptions.BadRequest: 400 Bad Request: functools.partial(<bound method FastApiDispatcher.dispatch of > <odoo.addons.fastapi.fastapi_dispatcher.FastApiDispatcher object at 0x7fec59b67c10>>), /app/department: Function declared as capable of handling request of type 'http' but called with a request of type 'json' 

**how to replicate**
Make a simple endpoint to create a new record (in this case I used the model `hr.department`) and try to make a call with json body (like in the docs page)

```python
class DepartmentInfo(BaseModel):
    id: int
    name: str

    class Config:
        orm_mode = True
        getter_dict = utils.GenericOdooGetter
    

class DepartmentRequest(BaseModel):
    name: str

@router.post("/department", response_model=DepartmentInfo)
def post_department(department: DepartmentRequest, env=Depends(odoo_env)) -> DepartmentInfo:
    dep = env["hr.department"].create(department)
    return DepartmentInfo.from_orm(dep)
```

**Comment by destroT** [comment]
My bad, I test it in a new database and It's a problem specific of the backport to odoo 14.0

**Comment by destroT** [comment]
Fixed with #356
